server/interceptor/interceptor.go

```diff
@@ -271,6 +271,38 @@ type rpcInfo struct {
 	tokens int
 }
 
+// chargable is satisfied by request proto messages which contain a GetChargeTo
+// accessor.
+type chargable interface {
+	GetChargeTo() *trillian.ChargeTo
+}
+
+// addChargedUsers adds quota spec entries for any chargable user quotas.
+// Quota specs are only added if req is non-nil and satisfies the chargable
+// interface.
+func (i *rpcInfo) addChargedUsers(req interface{}) {
+	if req == nil {
+		return
+	}
+	c, ok := req.(chargable)
+	if !ok {
+		return
+	}
+	chargeTo := c.GetChargeTo()
+	if chargeTo == nil {
+		return
+	}
+
+	kind := quota.Read
+	if !i.readonly {
+		kind = quota.Write
+	}
+
```

Comment:
Can we fold all the token-spec generation code together please?  (This and the `if info.tokens > 0` arm of `newRPCInfo()`)

Reply:
done